### PR TITLE
Introduce a new Docker image for data-pipeline spawning

### LINKED ISSUE #309: Use a dedicated Docker image for spawning the data-pipeline
See https://github.com/mozilla/bugbug/pull/293#discussion_r278443964.

We should have a dedicated Docker image with all the dependencies to stop downloading them each time.

### LINKED ISSUE #326: Taskcluster hook definition is using the 'data-pipeline-definition' branch
It should use the branch related to the latest released tag.

### PR DESCRIPTION
**Pull request by Lothiraldan**
I am not sure if the hook definition will like not having a command and if it will correctly takes the command from the image.

Fixes #309 and fixes #326.

### CODE REVIEW

File: infra/dockerfile.spawn_data_pipeline [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#syntax=docker/dockerfile:experimental
+FROM python:3.7.3-slim
+
+# Setup dependencies in a cachable step
+RUN pip install --disable-pip-version-check --no-cache-dir taskcluster==7.0.1 json-e==3.0.0 requests==2.21.0 pyyaml==5.1
```
Comment (marco-c) [inline infra/dockerfile.spawn_data_pipeline]: Maybe add the requirements in a requirements file, so pyup will update them?

File: infra/dockerfile.spawn_data_pipeline [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#syntax=docker/dockerfile:experimental
```
Comment (marco-c) [inline infra/dockerfile.spawn_data_pipeline]: We don't need the experimental here, right?

File: infra/dockerfile.spawn_data_pipeline [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+FROM python:3.7.3-slim
+
+# Setup dependencies in a cachable step
+RUN pip install --disable-pip-version-check --no-cache-dir taskcluster==7.0.1 json-e==3.0.0 requests==2.21.0 pyyaml==5.1
+
+ADD taskcluster-hook-pipeline-start.json /code
```
Comment (marco-c) [inline infra/dockerfile.spawn_data_pipeline]: This should be data-pipeline.yml, which should be passed to spawn_data_pipeline.py

**Comment by marco-c** [comment]
> I am not sure if the hook definition will like not having a command and if it will correctly takes the command from the image.

Yes, that should work!

File: infra/dockerfile.spawn_data_pipeline (hunk 2 above)
Comment (La0) [reply infra/dockerfile.spawn_data_pipeline]: Yeap: there are no `RUN --mount` commands

File: infra/dockerfile.spawn_data_pipeline [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+# Setup dependencies in a cachable step
+ADD spawn_data_pipeline_requirements.txt /code/
+
+RUN pip install --disable-pip-version-check --no-cache-dir -r /code/spawn_data_pipeline_requirements.txt
+
+ADD spawn_data_pipeline.py /code
+
+ADD data-pipeline.yml /code
+
+CMD python /code/spawn_data_pipeline /code/data-pipeline.yml
```
Comment (marco-c) [inline infra/dockerfile.spawn_data_pipeline]: `/code/spawn_data_pipeline` should be `/code/spawn_data_pipeline.py`, right?

**Pull request merged**